plugin/hooks.go

```diff
@@ -139,6 +140,11 @@ type Hooks interface {
 	// If actor is not nil, the user was removed from the team by the actor.
 	UserHasLeftTeam(c *Context, teamMember *model.TeamMember, actor *model.User)
 
+	// HealthCheck is invoked periodically to assert the plugin's current health.
+	// If the health check fails more than the configured number of times, the process will be destroyed and restarted.
+	//
+	HealthCheck() error
```

Comment:
Proposed rewording for docs:

> HealthCheck is invoked every 30 seconds to assert the plugin's current health.
If the health check fails, the process will be destroyed and restarted.
If the health check fails more than three times within one hour, the plugin will be deactivated until an admin chooses to restart it.


We could choose to be less specific with describing the criteria of deactivating the plugin. Though, I'm thinking that it would be good to know exactly how the health check interval and restart policy works from the perspective of a plugin author. Once there is a consensus on the values chosen, then the wording can be finalized.